extension/user/registerUser.js

```diff
@@ -0,0 +1,55 @@
+const uuidv4 = require('uuid/v4')
+const crypto = require('crypto')
+/**
+ * @typedef {Object} RegisterInputArgs
+ * @property {string} mail
+ * @property {string} password
+ * @property {string} firstName
+ * @property {string} lastName
+ * @property {?string} gender
+ * @property {?string} birthday
+ * @property {?string} phone
+ */
+
+/**
+ * Validation and sanitation is done at previous step
+ * @param {SDKContext} context
+ * @param {RegisterInputArgs} input
+ * @param {function} cb
+ */
+module.exports = (context, input, cb) => {
+  // is logged in
+  if (context.meta.userId) {
+    return cb(null, {
+      userId: context.meta.userId
+    })
+  }
+
+  const password = crypto.createHash('md5').update(input.password).digest('hex')
```

Comment:
There are better algorithms, than md5 for securing user passwords. The used hash algorithm could be stored in the user data for maintaining backwards compatibility.